app/views/form_drafts/show.haml

```diff
@@ -13,13 +13,16 @@
   = form_tag controller: :forms, action: :submit, id: @draft.form.id do
     = fields_for :user do |u|
       .field
-        .heading Please complete the following personal information.
-        //.explanation You will be able to log in and review your submission.
+        .explanation 
+          This is the personal information we have on file for you.
+          Please fill in any missing or incorrect information.
+          The confirmation will be sent to this email address.
       - [:first_name, :last_name, :email].each do |field|
         .field
           .label= u.label field
-          = u.text_field field, size: 50, required: true,
-            value: @current_user.try(field)
+          = u.text_field field, size: 50,
+            value: @current_user.try(field) || @placeholder.try(field),
```

Comment:
I agree with @dfaulken, although it makes this part simple to write and read.

It seems like you *could*  look at the session variables in the second clause of the `||` here:

```haml
- {first_name: 'givenName', last_name: 'surName', email: 'mail'}.each do |field, fallback|
  .field
    .label= u.label field
    = u.text_field field, size: 50,
      value: @current_user.try(field) || request.env[fallback]
```